src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,682 @@
+    /// Constructor.
+    pub fn new(scene:&Scene, number_of_ports:NonZeroU32) -> Self {
+        let network = default();
+        let frp     = Frp::new(&network);
+        let data    = OutputPortsData::new(scene.clone_ref(), number_of_ports);
+        let data    = Rc::new(data);
+        OutputPorts {data,network,frp}.init()
+    }
+
+    fn init(mut self) -> Self {
+        self.init_frp();
+        self
+    }
+
+    fn init_frp(&mut self) {
+        let network = &self.network;
+        let frp     = &self.frp;
+        let data    = &self.data;
+
+        // Used to set and detect the end of the tweens. The actual value is irrelevant, only the
+        // duration of the tween matters and that this value is reached after that time.
+        const TWEEN_END_VALUE:f32 = 1.0;
+
+        // Timer used to measure whether the hover has been long enough to show the ports.
+        let delay_show = Tween::new(&network);
+        delay_show.set_duration(SHOW_DELAY_DURATION);
+
+        // Timer used to measure whether the mouse has been gone long enough to hide all ports.
+        let delay_hide = Tween::new(&network);
+        delay_hide.set_duration(HIDE_DELAY_DURATION);
+
+        frp::extend! { network
+
+            // === Size Change Handling == ///
+
+            eval frp.set_size ((size) data.set_size(*size));
+
+
+            // === Hover Event Handling == ///
+
+            port_mouse_over <- source::<PortId>();
+            port_mouse_out  <- source::<PortId>();
+
+            delay_show_finished    <- delay_show.value.map(|t| *t>=TWEEN_END_VALUE );
+            delay_hide_finished    <- delay_hide.value.map(|t| *t>=TWEEN_END_VALUE );
+            on_delay_show_finished <- delay_show_finished.gate(&delay_show_finished).constant(());
+            on_delay_hide_finished <- delay_hide_finished.gate(&delay_hide_finished).constant(());
+
+            visible                <- delay_show_finished.map(|v| *v);
+
+            mouse_over_while_inactive  <- port_mouse_over.gate_not(&visible).constant(());
+            mouse_over_while_active    <- port_mouse_over.gate(&visible).constant(());
+
+            eval_ mouse_over_while_inactive ([delay_show,delay_hide]{
+                delay_hide.stop();
+                delay_show.reset();
+                delay_show.set_target_value(TWEEN_END_VALUE);
+            });
+            eval_ port_mouse_out ([delay_hide,delay_show]{
+                delay_show.stop();
+                delay_hide.reset();
+                delay_hide.set_target_value(TWEEN_END_VALUE);
+            });
+
+            activate_ports <- any(mouse_over_while_active,on_delay_show_finished);
+            eval_ activate_ports (delay_hide.stop_and_rewind());
+
+            activate_ports_with_selected <- port_mouse_over.sample(&activate_ports);
+
+            hide_all <- on_delay_hide_finished.map(f_!(delay_show.stop_and_rewind()));
+
+        }
+
+        let port_frp = PortFrp {network         : network.clone_ref(),
```

Comment:
never, ever do it, pls. Netwrok has to be owned by 1 thing, not multiple.

Reply:
I've changed it. But we should also consider changing the ownership semantics of `Network` then to disallow cloning general to prevent this from ever happening.